eng/scripts/Repo-File-Content-Replacements.ps1

```diff
@@ -0,0 +1,87 @@
+<#
+.SYNOPSIS
+The script is to do file content replacements under current folder.
+
+.DESCRIPTION
+We are having the script to help search all occurances of specified strings within scope and replace to the target strings.
```

Comment:
I understand the idea here is that if I need to do some string replacements across all our language repos, I can use this script, instead of checking out locally all 9 repos and doing textual search and replace.

However, it is not exactly clear to me right now how I should use this script. I think it would be great if this description (or a README accompanying this script) explains how to use it.

Basically, I am looking for a something like this:

> How to use this script to replace strings in all language repos?
>
> 1. Run the `link here to the pipeline` pipeline, providing as input `migration.yml`, which is to contain set of all the replacements you want to make. See `link to file` for an example you can adapt.
> 2. The pipeline will execute the string-find-and-replace logic on all our 9 language repos according to the `migration.yml` you provided as input, and create pull requests against all the repos asking you to review and approve the changes.
> 3. Review and approve all the PRs.

Please note I made up the instructions, because I am not exactly sure how to use the script.

Now that I think to it, these instructions probably should be mentioned in the header comment of the pipeline source (`migration.yml`). This script could just link to it.

Reply:
Sync with Wes, this pipeline is specific for agent pool migration. If we have generic use in future, we can pull out the common logic. We renamed it to `agent-pool-migration.yml`. The PR title and branch name are also related to the concept of migration.